datalake/src/main/java/com/sequenceiq/datalake/flow/create/handler/StorageConsumptionCollectionSchedulingHandler.java

```diff
@@ -0,0 +1,62 @@
+package com.sequenceiq.datalake.flow.create.handler;
+
+import static com.sequenceiq.cloudbreak.common.exception.NotFoundException.notFound;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.common.event.Selectable;
+import com.sequenceiq.datalake.flow.create.SdxCreateHandlerSelectors;
+import com.sequenceiq.datalake.flow.create.event.SdxCreateFailedEvent;
+import com.sequenceiq.datalake.flow.create.event.StorageConsumptionCollectionSchedulingRequest;
+import com.sequenceiq.datalake.flow.create.event.StorageConsumptionCollectionSchedulingSuccessEvent;
+import com.sequenceiq.datalake.repository.SdxClusterRepository;
+import com.sequenceiq.datalake.service.consumption.ConsumptionService;
+import com.sequenceiq.flow.reactor.api.handler.ExceptionCatcherEventHandler;
+import com.sequenceiq.flow.reactor.api.handler.HandlerEvent;
+
+import reactor.bus.Event;
+
+@Component
+public class StorageConsumptionCollectionSchedulingHandler extends ExceptionCatcherEventHandler<StorageConsumptionCollectionSchedulingRequest> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StorageConsumptionCollectionSchedulingHandler.class);
+
+    private final SdxClusterRepository sdxClusterRepository;
+
+    private final ConsumptionService consumptionService;
+
+    public StorageConsumptionCollectionSchedulingHandler(SdxClusterRepository sdxClusterRepository, ConsumptionService consumptionService) {
+        this.sdxClusterRepository = sdxClusterRepository;
+        this.consumptionService = consumptionService;
+    }
```

Comment:
please use field injection if possible, usually we are using that in datalake service

Reply:
Understood. All the datalake services / components touched in the PR have been modified in a similar way (and their UT updated accordingly).